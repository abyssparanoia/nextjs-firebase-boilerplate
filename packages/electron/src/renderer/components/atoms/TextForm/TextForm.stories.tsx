import React from 'react'
import { storiesOf } from '@storybook/react'
import { TextForm } from './TextForm'

storiesOf('atoms/TextForm', module).add('Required', () => <TextForm required label="Required" defaultValue="Value" />)

storiesOf('atoms/TextForm', module).add('Disabled', () => <TextForm disabled label="Disabled" defaultValue="Value" />)

storiesOf('atoms/TextForm', module).add('Password', () => (
  <TextForm label="Password" type="password" autoComplete="current-password" />
))

storiesOf('atoms/TextForm', module).add('number', () => <TextForm label="Number" type="number" defaultValue={3} />)

storiesOf('atoms/TextForm', module).add('Search Field', () => <TextForm label="Search field" type="search" />)

storiesOf('atoms/TextForm', module).add('Read Only', () => (
  <TextForm readOnly label="label" defaultValue="Value" helperText="Read Only" />
))

storiesOf('atoms/TextForm', module).add('Error', () => <TextForm error label="Error" defaultValue="Value" />)

storiesOf('atoms/TextForm', module).add('With Icon', () => (
  <TextForm required label="Required" defaultValue="Value" iconVariant="sample" />
))
