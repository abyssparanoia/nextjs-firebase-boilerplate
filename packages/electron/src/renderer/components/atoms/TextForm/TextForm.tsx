import React from 'react'
import { TextField } from '@material-ui/core'
import InputAdornment from '@material-ui/core/InputAdornment'
import { Icon, Variant as IconVariant } from '../../nano/Icon'

export interface TextFormProps {
  value?: string | number
  name?: string
  required?: boolean
  disabled?: boolean
  error?: boolean
  readOnly?: boolean
  label?: string
  type?: 'password' | 'search' | 'number'
  defaultValue?: string | number
  autoComplete?: string
  helperText?: string
  placeholder?: string
  iconVariant?: IconVariant
  iconCursor?: 'pointer'
  fullWidth?: boolean
  onChange?: (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void
  onClickIcon?: () => void
}

export const TextForm: React.FC<TextFormProps> = ({
  value,
  name,
  required,
  disabled,
  error,
  readOnly,
  label,
  type,
  defaultValue,
  autoComplete,
  helperText,
  iconVariant,
  iconCursor,
  placeholder,
  fullWidth,
  onChange,
  onClickIcon
}) => {
  return (
    <TextField
      value={value}
      name={name}
      required={required}
      disabled={disabled}
      error={error}
      label={label}
      type={type}
      placeholder={placeholder}
      defaultValue={defaultValue}
      autoComplete={autoComplete}
      helperText={helperText}
      fullWidth={fullWidth}
      onChange={onChange}
      InputProps={{
        readOnly,
        endAdornment: iconVariant && (
          <InputAdornment position="start" style={{ cursor: iconCursor }} onClick={onClickIcon}>
            <Icon width={24} height={24} variant={iconVariant} />
          </InputAdornment>
        )
      }}
    />
  )
}
