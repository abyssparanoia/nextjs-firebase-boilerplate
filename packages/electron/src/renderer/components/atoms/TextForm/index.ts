export { TextForm } from './TextForm'
export * from './TextForm'
