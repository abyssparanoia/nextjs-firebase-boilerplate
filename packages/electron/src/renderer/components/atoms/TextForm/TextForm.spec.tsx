import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import { TextForm } from './TextForm'

describe(`${TextForm.name}`, () => {
  test('Snapshot', () => {
    const component = render(<TextForm value="value" />)
    const tree = component.baseElement
    expect(tree).toMatchSnapshot()
  })

  test('The onChange fires when the Change Event fires.', () => {
    const onChange = jest.fn()
    const VALUE = 'value'
    render(<TextForm value={VALUE} onChange={onChange} />)
    fireEvent.change(screen.getByDisplayValue(VALUE), {
      target: { value: 'chuck' }
    })
    expect(onChange).toHaveBeenCalled()
  })

  test('The placeholder is working properly.', () => {
    const PLACEHOLDER = 'placeholder'
    render(<TextForm value={PLACEHOLDER} />)
    expect(screen.getByDisplayValue(PLACEHOLDER))
  })
})
