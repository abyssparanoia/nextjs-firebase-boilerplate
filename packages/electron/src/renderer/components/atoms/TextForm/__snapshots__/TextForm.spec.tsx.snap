// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          type="text"
          value="value"
        />
      </div>
    </div>
  </div>
</body>
`;
